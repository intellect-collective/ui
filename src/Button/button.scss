@mixin button-variant($color, $highlight, $text) {
    background-color: $color;
    color: text;

    &:hover {
        background-color: $highlight;
    }

    &-outline {
        background-color: transparent;
        border: 1px solid $color;
        color: $color;

        &:hover {
            background-color: $highlight;
            border-color: $highlight;
            color: $text;
        }
    }
}

@if (null != index($ui-includes, 'form.button') and null == index($ui-excludes, 'form.button')) {
    $button-colors: $theme !default;

    .btn {
        background-color: transparent;
        border: none;
        border-radius: 2px;
        color: $color-white;
        cursor: pointer;
        display: inline-block;
        font-size: 1rem;
        height: 36px;
        letter-spacing: .5px;
        line-height: 36px;
        outline: 0;
        padding: 0 2rem;
        text-align: center;
        text-decoration: none;
        text-transform: uppercase;
        transition: .2s ease-out;
        vertical-align: middle;
        -webkit-tap-highlight-color: transparent;

        &:hover {
            background-color: transparent;
            text-decoration: none;
        }

        &:not(.btn-flat) {
            box-shadow: 0 2px 2px 0 rgba(0,0,0,0.14), 0 1px 5px 0 rgba(0,0,0,0.12), 0 3px 1px -2px rgba(0,0,0,0.2);

            &:hover {
                box-shadow: 0 3px 3px 0 rgba(0,0,0,0.14), 0 1px 7px 0 rgba(0,0,0,0.12), 0 3px 1px -1px rgba(0,0,0,0.2);
            }
        }

    }

    .btn-small {
        font-size: .85rem;
        height: 30px;
        line-height: 30px;
        padding: 0 1rem;
    }

    .btn-large {
        height: 54px;
        line-height: 54px;
    }

    @each $name, $attributes in $button-colors {
        .btn-#{$name} {
            @include button-variant(map-get($attributes, 'color'), map-get($attributes, 'highlight'), map-get($attributes, 'text'));
        }
    }
}