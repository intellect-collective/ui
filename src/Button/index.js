import React from 'react';
import clsx from 'clsx';

const Button = ({ color, size, shape, className, classes, ...rest }) => (
	<button { ...rest }
			className={ clsx(className, classes, 'btn', color, size, shape )} />
);
export default Button;
