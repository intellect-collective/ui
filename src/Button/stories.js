import React from 'react';
import cartesian from 'storybook-cartesian';
import { storiesOf } from '@storybook/react';
import { Button } from '../..';

storiesOf('2 - Components / Form / Button', module)
    .add('Colors', () => (
    	<div>
    		<Button color="primary">Primary</Button>
    		<Button color="secondary">Secondary</Button>
    		<Button color="tertiary">Tertiary</Button>
    		<Button color="success">Success</Button>
    		<Button color="info">Info</Button>
    		<Button color="warning">Warning</Button>
    		<Button color="error">Error</Button>
    		<Button>Default</Button>
    	</div>
	))
	.add('Sizes', () => (
    	<div>
    		<Button size="xsmall">X-Small</Button>
    		<Button size="small">Small</Button>
    		<Button size="medium">Medium</Button>
    		<Button>Default</Button>
    		<Button size="large">Large</Button>
    		<Button size="xlarge">X-Large</Button>
    	</div>
	))
	.add('Shape', () => (
    	<div>
    		<Button>Default</Button>
    		<Button shape="square">Square</Button>
    		<Button shape="rounded">Rounded</Button>
    		<Button shape="circle">Circle</Button>
    	</div>
	))
	.add('Contents', () => (
    	<div>
    		<Button>Click Me</Button>
    		<Button>你好</Button>
    	</div>
	));
