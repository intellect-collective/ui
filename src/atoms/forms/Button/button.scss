@import 'mixins';

@if (null != index($ui-includes, 'form.button') and null == index($ui-excludes, 'form.button')) {
    $btn-font-weight: normal !default;
    $btn-default-color: #333 !default;
    $cursor-disabled: disabled !default;

    $btn-border-radius-base:    $border-radius-base !default;
    $btn-border-radius-large:   $border-radius-large !default;
    $btn-border-radius-small:   $border-radius-small !default;

    $btn-default-color:         #333 !default;
    $btn-default-bg:            #fff !default;
    $btn-default-border:        #ccc !default;

    $btn-primary-color:         text-color($brand-primary) !default;
    $btn-primary-bg:            $brand-primary !default;
    $btn-primary-border:        darken($btn-primary-bg, 5%) !default;

    $btn-success-color:         text-color($brand-success) !default;
    $btn-success-bg:            $brand-success !default;
    $btn-success-border:        darken($btn-success-bg, 5%) !default;

    $btn-info-color:            text-color($brand-info) !default;
    $btn-info-bg:               $brand-info !default;
    $btn-info-border:           darken($btn-info-bg, 5%) !default;

    $btn-warning-color:         text-color($brand-warning) !default;
    $btn-warning-bg:            $brand-warning !default;
    $btn-warning-border:        darken($btn-warning-bg, 5%) !default;

    $btn-danger-color:          text-color($brand-danger) !default;
    $btn-danger-bg:             $brand-danger !default;
    $btn-danger-border:         darken($btn-danger-bg, 5%) !default;

    $btn-sizes: (
        '.btn': ($padding-base-vertical, $padding-base-horizontal, $font-size-base, $line-height-base, $btn-border-radius-base),
        '.btn-xs': ($padding-xs-vertical, $padding-xs-horizontal, $font-size-small, $line-height-small, $btn-border-radius-small),
        '.btn-sm': ($padding-small-vertical, $padding-small-horizontal, $font-size-small, $line-height-small, $btn-border-radius-small),
        '.btn-lg': ($padding-large-vertical, $padding-large-horizontal, $font-size-large, $line-height-large, $btn-border-radius-large)
    ) !default;
    $btn-variants: (
        ('.btn', '.btn-default'): ($btn-default-color, $btn-default-bg, $btn-default-border),
        ('.btn-primary', '.btn-brand'): ($btn-primary-color, $btn-primary-bg, $btn-primary-border),
        ('.btn-success', '.btn-green'): ($btn-success-color, $btn-success-bg, $btn-success-border),
        ('.btn-info', '.btn-alert', '.btn-blue'): ($btn-info-color, $btn-info-bg, $btn-info-border),
        ('.btn-warning', '.btn-orange'): ($btn-warning-color, $btn-warning-bg, $btn-warning-border),
        ('.btn-danger', '.btn-error', '.btn-red'): ($btn-danger-color, $btn-danger-bg, $btn-danger-border)
    ) !default;

    .btn {
        background-image: none;
        border: 1px solid transparent;
        display: inline-block;
        cursor: pointer;
        font-weight: $btn-font-weight;
        margin-bottom: 0;
        text-align: center;
        touch-action: manipulation;
        transition: .2s ease-out;
        user-select: none;
        vertical-align: middle;
        white-space: nowrap;

        &,
        &:active,
        &.active {
            &:focus,
            &.focus {
                @include tab-focus();
            }
        }

        &:hover,
        &:focus,
        &.focus {
            color: $btn-default-color;
            text-decoration: none;
        }

        &:active,
        &.active {
            outline: 0;
            background-image: none;
            box-shadow: inset 0 3px 5px rgba(0,0,0,.125);
        }

        &.disabled,
        &[disabled],
        fieldset[disabled] & {
            cursor: $cursor-disabled;
            opacity: .65;
            box-shadow: none;
        }

        &a {
            &.disabled,
            fieldset[disabled] & {
                pointer-events: none;
            }
        }
    }

    @each $class, $attributes in $btn-sizes {
        #{$class} {
            @include button-size($attributes...);
        }
    }

    @each $classes, $attributes in $btn-variants {
        @each $class in $classes {
            #{$class} {
                @include button-variant($attributes...);
            }
        }
    }
}


