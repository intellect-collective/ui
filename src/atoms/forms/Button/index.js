import React from 'react';
import PropTypes from 'prop-types';
import clickable from '../_decorators/clickable';
import formContext from '../_decorators/formContext';
import { compose } from '../_decorators/_utils';

const Button = (props) => (
    <button { ...props } />
);
Button.displayName = 'Button';
Button.propTypes = {
    type: PropTypes.oneOf(['button', 'reset', 'submit']),
    value: PropTypes.string
};
Button.defaultProps = {
    type: 'button',
    value: ''
};

export default compose(
    formContext(),
    clickable()
)(Button);
