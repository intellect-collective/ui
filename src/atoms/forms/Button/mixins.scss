@function alter-color($color, $amount: 10%) {
    @if (lightness($color) > 50) {
        @return darken($color, $amount);
    }
    @return lighten($color, $amount);
}

@mixin button-variant($color, $background, $border: $background) {
    color: $color;
    background-color: $background;
    border-color: $border;

    &:focus,
    &.focus {
        color: $color;
        background-color: alter-color($background, 10%);
        border-color: alter-color($border, 25%);
    }
    &:hover {
        color: $color;
        background-color: alter-color($background, 10%);
        border-color: alter-color($border, 12%);
    }
    &:active,
    &.active,
    .open > .dropdown-toggle#{&} {
        color: $color;
        background-color: alter-color($background, 10%);
        background-image: none;
        border-color: alter-color($border, 12%);

        &:hover,
        &:focus,
        &.focus {
            color: $color;
            background-color: alter-color($background, 17%);
            border-color: alter-color($border, 25%);
        }
    }
    &.disabled,
    &[disabled],
    fieldset[disabled] &,
    fieldset.disabled & {
        color: lighten($color, 10%);
        background-color: lighten($background, 10%)
        &:hover,
        &:focus,
        &.focus {
            background-color: $background;
            border-color: $border;
        }
    }

    .badge {
        color: $background;
        background-color: $color;
    }
}

@mixin button-size($padding-vertical, $padding-horizontal, $font-size, $line-height, $border-radius) {
    padding: $padding-vertical $padding-horizontal;
    font-size: $font-size;
    line-height: $line-height;
    border-radius: $border-radius;
}
