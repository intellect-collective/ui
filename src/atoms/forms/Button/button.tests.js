import React from 'react';
import Button from '.';
import {
    getInvalidFieldError,
    basicClickableTests
} from '../../../../test/utils';

const validProps = {
    name: 'test',
    type: 'submit',
    onClick: () => {}
};

const expectedProps = {
    name: 'test',
    type: 'submit',
    onClick: expect.any(Function)
};

describe('Button', () => {
    basicClickableTests(Button, 'button', validProps, expectedProps);

    it.only('Should throw an error on invalid type', () => {
        expect(() => {
            mount(<Button name="test" type="invalid">Execute</Button>);
        }).toThrowError(getInvalidFieldError('type'));
    });

    it('Should have a default type', () => {
        const wrapper = mount(<Button name="test" value="Execute" />);
        expect(wrapper.find('button').prop('type')).toEqual('button');
    });
});
