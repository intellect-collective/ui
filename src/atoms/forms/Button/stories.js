import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import Button from './index';

storiesOf('Atoms / Forms / Button', module)
	.add('button element', () => (
		<div>
			<Button className="btn">Default</Button>
			<Button className="btn btn-primary">btn-primary</Button>
			<Button className="btn btn-success">btn-success</Button>
			<Button className="btn btn-info">btn-info</Button>
			<Button className="btn btn-warning">btn-warning</Button>
			<Button className="btn btn-danger">btn-danger</Button>
		</div>
	))
	.add('other elements', () => (
		<div>
			<a className="btn btn-primary" href="#" role="button">Link</a>
			<button className="btn btn-primary" type="submit">Button</button>
			<input className="btn btn-primary" type="button" value="Input" />
			<input className="btn btn-primary" type="submit" value="Submit" />
			<input className="btn btn-primary" type="reset" value="Reset" />
		</div>
	))
	.add('outline', () => (
		<div>
			<button type="button" class="btn btn-primary btn-outline">btn-outline-primary</button>
			<button type="button" class="btn btn-success btn-outline">btn-outline-success</button>
			<button type="button" class="btn btn-info btn-outline">btn-outline-info</button>
			<button type="button" class="btn btn-warning btn-outline">btn-outline-warning</button>
			<button type="button" class="btn btn-danger btn-outline">btn-outline-danger</button>
		</div>
	));
