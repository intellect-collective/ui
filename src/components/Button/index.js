import React from 'react';
import './button.scss';

export default (props) => (
	<button { ...props } />
);
