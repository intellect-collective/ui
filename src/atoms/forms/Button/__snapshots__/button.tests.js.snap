// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Basic clickable tests Should accept valid props 1`] = `
<FormContext
  name="test"
  onClick={[Function]}
  type="submit"
>
  <FormContextInner
    ctx={null}
    name="test"
    onClick={[Function]}
    type="submit"
  >
    <Clickable
      name="test"
      onClick={[Function]}
      type="submit"
    >
      <Button
        name="test"
        onClick={[Function]}
        type="submit"
        value=""
      >
        <button
          name="test"
          onClick={[Function]}
          type="submit"
          value=""
        />
      </Button>
    </Clickable>
  </FormContextInner>
</FormContext>
`;
